pkg/test/env/ports.go

```diff
@@ -0,0 +1,150 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package env
+
+import (
+	"log"
+)
+
+// Dynamic port allocation scheme
+// In order to run the tests in parallel. Each test should use unique ports
+// Each test has a unique test_name, its ports will be allocated based on that name
+
+// All tests should be listed here to get their test ids
+const (
+	FailedRequestTest uint16 = iota
+	FaultInjectTest
+	NetworkFailureTest
+	XDSTest
+	IstioAuthnTestOriginRejectNoJwt
+	IstioAuthnTestPeerRejectNoJwt
+	IstioAuthnTestPeerRejectNoMtls
+	IstioAuthnTestPeerRejectNoTLS
+	RouteDirectiveTest
+	DynamicAttributeTest
+	DynamicListenerTest
+	PilotPluginTest
+	PilotPluginTCPTest
+	PilotPluginTLSTest
+	PilotMCPTest
+	RbacGlobalPermissiveTest
+	RbacPolicyPermissiveTest
+	GatewayTest
+	SidecarTest
+	SidecarConsumerOnlyTest
+	TracingHeaderTest
+	STSTest
+	STSCacheTest
+	STSRenewTest
+	STSFailureTest
+	STSTimeoutTest
+	STSServerCacheTest
+	STSShortLivedCacheTest
+	SDSTest
+	SDSCertRotation
+	CSRFailure
+	BadCSRResponse
+
+	// The number of total tests. has to be the last one.
+	maxTestNum
+)
+
+const (
+	portBase uint16 = 20000
+	// Maximum number of ports used in each test.
+	portNum uint16 = 8
+	// Number of ports used by Envoy in each test.
+	envoyPortNum uint16 = 4
+)
+
+// Ports stores all used ports
+type Ports struct {
+	ClientProxyPort uint16
+	ServerProxyPort uint16
+	TCPProxyPort    uint16
+	AdminPort       uint16
+	BackendPort     uint16
+	DiscoveryPort   uint16
+	STSPort         uint16
+	ExtraPort       uint16
```

Comment:
Can you add a comment to describe "ExtraPort" (e.g., what this port is used for)?